packages/ove-app-networks/src/client/common/networks.js

```diff
@@ -13,6 +13,235 @@ $(function () {
     });
 });
 
+runOperation = function (message) {
+    // Helper method to retrieve a property from an element
+    const getFromElement = function (e, x) {
+        let m;
+        let n;
+        if (!x || x.indexOf('.') === -1) {
+            n = x;
+        } else {
+            m = x.substring(x.indexOf('.') + 1);
+            n = x.substring(0, x.indexOf('.'));
+        }
+        const p = e[n] || (e.attributes ? e.attributes[n] : undefined);
+        return m ? getFromElement(p, m) : p;
+    };
+
+    // Helper method to retrieve a property from a message
+    const getFromMessage = function (message, p1, p2) {
+        return message[p1] ? message[p1][p2] : undefined;
+    };
+
+    // IMPORTANT: There are no logs within the filter evaluation operations to ensure
+    // the most optimum performance. The logs corresponding to the operation carried out
+    // can be used for debugging purposes.
+    const evaluate = function (element, filter) {
+        // Evaluate as a number
+        const evaluateN = function (element, filter) {
+            return +(evaluate(element, filter));
+        };
+        // Evaluate as a string
+        const evaluateS = function (element, filter) {
+            return evaluate(element, filter).toString();
+        };
+        // Evaluate a function
+        const evaluateF = function (element, func, args) {
+            switch (func) {
```

Comment:
could the switch be reordered to group by #operands? then argument length checking may be done per batch?

Reply:
That unwantedly complicates the entire implementation and makes it very hard to follow. These are all string operations, and therefore no further grouping would be required. The only reason why we have argument length checking for the substring operation is because the spec allows two variants of it.

Reply:
I don't think we need to re-order the switch statement, but I think there is a benefit to having some logic based on the arity of the function.

The main thing making the current code hard to read is the frequent repetition of `evaluateS(element, args[0])` everywhere, with the occasional `evaluateS(element, args[1])` and `evaluateS(element, args[2])`.

This could be avoided by something like:

```JavaScript
const firstArg = evaluateS(element, args[0]);

if (args.length > 1){
   const secondArg = evaluateS(element, args[1]);
}

if (args.length > 2){
   const thirdArg = evaluateS(element, args[2]);
}
```

Reply:
@jamesscottbrown good plan, that solves both concerns